File: ios/CopyToClipboard.m
//
//  CopyToClipboard.m
//  MultipleEnvironment
//
//  Created by Semih Çakmak on 24.03.2023.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTLog.h>


@interface RCT_EXTERN_MODULE(CopyToClipboard, NSObject)

RCT_EXTERN_METHOD(funcWithParams: (NSString *)text second:(NSNumber *) second)
RCT_EXTERN_METHOD(basicFunc)
RCT_EXTERN_METHOD(basicFuncCallback:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(asyncFunc:(NSNumber *)num name:(NSString *)name surname:(NSString *)surname resolve:(RCTPromiseResolveBlock)resolve    reject:(RCTPromiseRejectBlock)reject   )
RCT_EXTERN_METHOD(copy:(NSString *)text)
RCT_EXTERN_METHOD(paste:(RCTResponseSenderBlock *)callback)
RCT_EXTERN_METHOD(funcWithOptionalParams:(NSString *)first second:(nonnull NSNumber * )second third:(nonnull NSNumber *)third dic:(NSDictionary *)dic )
@end

File: ios/CopyToClipboard.swift
//
//  CopyToClipboard.swift
//  MultipleEnvironment
//
//  Created by Semih Çakmak on 24.03.2023.
//

import Foundation


@objc(CopyToClipboard)
class CopyToClipboard : NSObject {
  
  @objc static func requiresMainQueueSetup() -> Bool { return true }
  
 @objc
  func funcWithParams(_ text:String,second:Int){
    print("copy func \(text)  ikinci param \(second)")
    
  }
  
 @objc
 func basicFunc() -> Void {
    print("basic func")
  }
  
  @objc func basicFuncCallback(_ callback:RCTResponseSenderBlock){
    callback(["semih",2,["1","2","3"]])
  }
  
  @objc func asyncFunc(_ num: Int, name : String, surname : String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock){
    
    if name == "sem" {
      resolve("haha data")
    } else {
      reject("2rejet","2. param",nil)
    }
   
  }
  
  @objc func funcWithOptionalParams(_ first:String?,second:NSNumber,third:NSNumber,dic:[String:Any]?){
//    guard let newDic = dic as? [String:Any] else{
//      return
//    }
    print("first:\(first ?? "ilk değer")  second:\(second )  third:\(third) dictionary: \(dic ?? ["default":1] )")
  }
  
  @objc func copy (_ text:String){
    UIPasteboard.general.string = text
  }
  
  @objc func paste (_ callback:RCTResponseSenderBlock){
    
    callback([UIPasteboard.general.string ?? ""])
    
  }
}
